config/routes.rb

```diff
@@ -280,8 +280,8 @@
   resource :profile, :only => [:edit, :update]
 
   # friendships
-  match "/user/:display_name/make_friend" => "friendships#make_friend", :via => [:get, :post], :as => "make_friend"
-  match "/user/:display_name/remove_friend" => "friendships#remove_friend", :via => [:get, :post], :as => "remove_friend"
+  match "/user/:display_name/follow_user" => "friendships#follow_user", :via => [:get, :post], :as => "follow_user"
+  match "/user/:display_name/unfollow_user" => "friendships#unfollow_user", :via => [:get, :post], :as => "unfollow_user"
```

Comment:
From a quick look at some gems that implement this functionality ([1](https://github.com/cmer/socialization), [2](https://github.com/tcocca/acts_as_follower)) the convention is to use "follow" as a noun, so table is `follows`, controller is `follows_controller`, model is `Follow` etc. But I think we should rename the controllers/models separately.

What it does suggest though is that we should use resourceful routing with this in mind, so for example "/user/:display_name/follow" with create/destroy actions (and new/edit for the GET pages). See `user_mutes` for an example in the routes.rb file that's a bit like this.

Reply:
When I started implementing same architecture as `user_mutes`, it caused to change some more files and add some complexities. This PR is already `+104 -104` line change. I think, it will be better to create a separate PR about changing routing architecture of the `follow` functionality after this PR will be merged to avoid making it even bigger and more complex.

Reply:
@AntonKhorev `app/views/user_mailer/friendship_notification.html.erb` and `app/views/user_mailer/friendship_notification.text.erb` changes are about redirections from mail notifications. For the consistent visual output this PR is focused on changing every occurrence of "friend" on user side and changing them to "following". Main focus is what will users see (including both texts and URLs). I agree that it will be great if friendship functionality is refactored (currently it is just renamed), but those changes are out of scope of this PR and will be done as a separate PR as soon as this PR will be merged (adding them to this PR will complicate things as there will be not only user side changes, but logical changes in `routes.rb`, `ability.rb` and etc. which will need much deeper testing than just text and URL comparisons).

Reply:
> `app/views/user_mailer/friendship_notification.html.erb` and `app/views/user_mailer/friendship_notification.text.erb` changes are about redirections from mail notifications.

I don't see any redirects from old GET paths to new GET paths (but I see changes in tests that checked the paths).

> For the consistent visual output this PR is focused on changing every occurrence of "friend" on user side and changing them to "following". Main focus is what will users see (including both texts and URLs).

Maybe it shouldn't focus on *both*?

Reply:
PR was updated accordingly to include refactor and renaming of the `friendship` functionality.